hibernate-core/src/main/java/org/hibernate/sql/ast/tree/from/NamedTableReference.java

```diff
@@ -72,7 +81,7 @@ public List<String> getAffectedTableNames() {
 
 	@Override
 	public boolean containsAffectedTableName(String requestedName) {
-		return isEmpty( requestedName ) || getTableExpression().contains( requestedName );
+		return isEmpty( requestedName ) || tableExpression.equals( requestedName );
```

Comment:
Actually, I think there are some other uses of `getTableExpression()` in this class that are probably also "wrong" in the sense that the result of using that method would lead to wrong results if the table reference has a pruned table expression. Can you please revisit the uses in this class and replace them with the `tableExpression` field?

Reply:
Right, I only replaces this occurrence because I had previously changed `equals` to `contains` for this reason. I've now substituted all uses of `getTableExpression()` inside the class, except for `getTableId()`, with direct field references.